### PR TITLE
fix: Update DotNetCore/Env/Versions

### PR DESCRIPTION
**Pull request by daffinito**
# Issue

https://github.com/newrelic/newrelic-diagnostics-cli/issues/111

# Goals

Update DotNetCore/Env/Versions to use `dotnet --info` for version info. fall back on `dotnet --version` then looking at dirs

# Implementation Details

using a basic enum of sources and loop through until we get versions back

# Testing

There were not tests for this task so I wrote some

`go test ./... -cover`

### CODE REVIEW

File: tasks/dotnetcore/env/versions.go [hunk 1]
```diff
@@ -68,22 +79,98 @@ func (t DotNetCoreEnvVersions) Execute(options tasks.Options, upstream map[strin
...
-	errorMessage := "Unable to complete this health check because we ran into some unexpected errors when attempting to collect this application's .NET Core SDK version:\n"
+func (t DotNetCoreEnvVersions) checkVersions(envVars map[string]string) ([]string, string) {
+	source := 0
+	errorMessages := ""
+	versions, errMessage, keepGoing := t.checkVersionWithSource(envVars, source)
+	for keepGoing {
+		source++
+		errorMessages += errMessage
+		versions, errMessage, keepGoing = t.checkVersionWithSource(envVars, source)
+	}
```
Comment (cade-conklin) [inline tasks/dotnetcore/env/versions.go:90]: The only thing I might change is calling this function outside of the function and inside it.  Maybe something like:
keepGoing := true
for keepGoing{
   versions, errMessage, keepGoing := t.checkVersionWithSource(envVars, source)
   errorMessages += errMessage
   source++
}

This isn't necessary to change it just makes more sense when I read the code

**Review by cade-conklin — DISMISSED** [review]
I approve!

File: tasks/dotnetcore/env/versions.go (hunk 1 above)
Comment (daffinito) [reply tasks/dotnetcore/env/versions.go]: Good call out! I've refactored it to be more readable (I hope)

**Review by cade-conklin — APPROVED** [review]
Looks good to me!

**Pull request merged**